edit/editor.go

```diff
@@ -708,41 +708,57 @@ func edit(ed *Editor, cmd []rune, w io.Writer) (addr, error) {
 	}
 }
 
-func parseText(cmd []rune) []rune {
-	switch {
-	case len(cmd) > 0 && cmd[0] == '\n':
-		return parseLines(cmd)
-	case len(cmd) > 0:
-		return parseDelimited(cmd[0], false, cmd[1:])
-	default:
+func parseText(e []rune) []rune {
+	var i int
+	for i < len(e) && unicode.IsSpace(e[i]) {
+		if e[i] == '\n' {
+			return parseLines(e[i+1:])
+		}
+		i++
+	}
+	if i == len(e) {
 		return nil
 	}
+	return parseDelimited(e[i], e[i+1:])
 }
 
-func parseLines(cmd []rune) []rune {
-	var rs []rune
-	for i := 1; i < len(cmd); i++ {
-		rs = append(rs, cmd[i])
-		if i < len(cmd)-1 && cmd[i] == '\n' && cmd[i+1] == '.' {
-			break
+func parseLines(e []rune) []rune {
+	var i int
+	var nl bool
+	for i = 0; i < len(e); i++ {
+		if nl && e[i] == '.' {
+			switch {
+			case i == len(e)-1:
```

Comment:
Ahh. I did it this way, because later each of the two switch cases returns something different. It's fine. This commit shouldn't be polluted with artifacts of some future change.